src/CMake/FindADIOS2.cmake

```diff
@@ -55,6 +61,12 @@ ELSE()
 ENDIF()
 
 SET_UP_THIRD_PARTY(ADIOS2 ${LIB} include adios2)
+THIRD_PARTY_INSTALL_LIBRARY(${VISIT_ADIOS2_DIR}/${LIB}/libadios2_atl.so)
```

Comment:
Is there a reason why you didn't just include all off the needed adios lib names in the call to SET_UP_THIRD_PARTY ?  
Use of .so as the libname extension is unix specific, so this won't work for the mac or windows.

Reply:
In the case of parallel, `SET_UP_THIRD_PARTY` is called twice. Once at the top and again at the bottom. Is that harmless? Does the latter call just wind up overriding the former? Would it be better to if-then-else this logic at the top?

Reply:
Regarding Kathleen's comment: I actually tried listing them all in SET_UP_THIRD_PARTY. It added adios2 with a full path to the "so" and the remaining libraries were added with "-l" without a corresponding "-L", so the linking failed since it couldn't find the library. For example, I listed "adios2 adios2_atl" and it generated "/really/long/path/to/adios2/library/libadios2.so -ladios2_atl". Is this a bug or am I using it wrong?

Reply:
Regarding Mark's comment: I see 2 calls to SET_UP_THIRD_PARTY, the first is with ADIOS2 and the second when parallel is also enabled is with ADIOS2_PAR. So I believe it should be called twice. Am I correct, or am I missing something?

Reply:
@brugger1 ... my apologies. I wasn't looking closely. I hadn't noticed each call is for a different Adios variant. Regarding @biagas original query...I think she might have been expecting this...

```
SET_UP_THIRD_PARTY(ADIOS2 ${LIB} include adios2 adios2_atl adios2_dill adios2_evpath adios2_ffs adios2_sst taustubs)
```

Reply:
@markcmiller86 is correct in what I was expecting.
I just realized that this plugin suffers from a similar deficiency in our use of CMake as MOAB regarding handling of separate TP lib locations for parallel and serial versions.
Our current setup requires LDFLAGS tag in the plugin's .xml file to specify library location and LIBS tag to have a list of library names to be found where LDFLAGS points.  This currently doesn't support per-component LDFLAGS  (the LIBS tag does).  I wonder if adding per-component support would fix the problem for ADIOS2?

Reply:
I changed the call to SET_UP_THIRD_PARTY as Kathleen suggested. I also had to modify ADIOS2.xml. I added LDFLAGS and MPI_LDFLAGS. This required running xml2cmake on the file. Since MPI_LDFLAGS isn't supported I actually manually updated it. I noticed that ADIOS2.xml had an MPI_CXXFLAGS which also isn't supported, so it was already getting manually generated, so it isn't really any worse now.

Reply:
I retested the building of a package, installing and running in serial and parallel with some ADIOS2 data from Dave Pugmire.